artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java

```diff
@@ -69,24 +68,16 @@ public boolean containsWildCard() {
 
    @Override
    public List<Address> getLinkedAddresses() {
-      return linkedAddresses == null ? Collections.emptyList() : linkedAddresses;
+      return linkedAddresses;
    }
 
    @Override
    public void addLinkedAddress(final Address address) {
-      if (linkedAddresses == null) {
-         linkedAddresses = new ArrayList<>(1);
-      }
-      if (!linkedAddresses.contains(address)) {
-         linkedAddresses.add(address);
-      }
+      linkedAddresses.add(address);
```

Comment:
I would add an `assert !linkedAddress,contains(address)` here as an invariant on the tests

Reply:
Dont add assert into runtime code.

Reply:
It is better to follow the pattern from set interface and return true / false if operation actually actioned